packages/flutter_tools/lib/src/pre_run_validator.dart

```diff
@@ -41,6 +46,14 @@ class _DefaultPreRunValidator implements PreRunValidator {
       );
     }
   }
+
+  // Delete any temp dirs created by previous tool invocations
+  void _deletePreviousToolTempDirs() {
+    // this is not relevant with test file systems
+    if (fileSystem is LocalFileSystem) {
```

Comment:
I think `fileSystem` will be an `ErrorHandlingFileSystem` here and not a `LocalFileSystem`. You could check with some debug prints, or manually create an entry `/tmp/flutter_tools...` and make sure it goes away.

Reply:
ahh, you're right

Reply:
updated the code and verified it locally